src/main/java/org/folio/rest/impl/PatronServicesResourceImpl.java

```diff
@@ -246,82 +254,111 @@ private User convertJsonToUser(JsonObject userJson) {
       } else {
-        return PostPatronAccountResponse.respond422WithTextPlain("User does not belong to the required patron group");
+        logger.warn("processSingleUser::User does not belong to the required patron group");
+        return PostPatronAccountResponse.respond422WithApplicationJson(createError("User does not belong to the required patron group", BAD_REQUEST_CODE));
       }
     });
   }
 
   private CompletableFuture<String> getRemotePatronGroupId(UserRepository userRepository, Map<String, String> okapiHeaders) {
+    logger.info("getRemotePatronGroupId::Attempting to retrieve remote patron group ID");
+
     return userRepository.getGroupByGroupName(REMOTE_GROUP, okapiHeaders)
       .thenApply(responseJson -> {
-        if (responseJson.getInteger(TOTAL_RECORDS) > 0) {
-          return responseJson.getJsonArray(USER_GROUPS).getJsonObject(0).getString(ID);
+        int totalRecords = responseJson.getInteger(TOTAL_RECORDS);
+        logger.info("getRemotePatronGroupId::Received response with total records: {}", totalRecords);
+
+        if (totalRecords > 0) {
+          String groupId = responseJson.getJsonArray(USER_GROUPS).getJsonObject(0).getString(ID_FILED);
+          logger.info("getRemotePatronGroupId::Found remote patron group ID: {}", groupId);
+          return groupId;
         } else {
-          throw new IllegalArgumentException(new HttpException(500, "Remote patron group not found"));
+          String errorMessage = "Remote patron group not found";
+          logger.error("getRemotePatronGroupId::{}", errorMessage);
+          throw new IllegalArgumentException(new HttpException(500, errorMessage));
         }
+      })
+      .exceptionally(throwable -> {
+        logger.error("getRemotePatronGroupId::Exception occurred while retrieving remote patron group ID", throwable);
+        throw new CompletionException(throwable);
       });
   }
 
   private CompletableFuture<JsonObject> getAddressTypes(UserRepository userRepository, Map<String, String> okapiHeaders) {
+    logger.info("getAddressTypes::Trying to get addressTypes");
     return userRepository.getAddressByType(okapiHeaders)
       .thenApply(responseJson -> {
-        JsonObject addressTypes = new JsonObject();
+        final JsonObject addressTypes = new JsonObject();
         responseJson.getJsonArray(ADDRESS_TYPES).forEach(item -> {
-          JsonObject addressType = (JsonObject) item;
-          if (HOME.equalsIgnoreCase(addressType.getString(ADDRESS_TYPE))) {
-            addressTypes.put(HOME, addressType.getString(ID));
-          } else if (WORK.equalsIgnoreCase(addressType.getString(ADDRESS_TYPE))) {
-            addressTypes.put(WORK, addressType.getString(ID));
+          final JsonObject addressType = (JsonObject) item;
+          if (HOME_FIELD.equalsIgnoreCase(addressType.getString(ADDRESS_TYPE_FIELD))) {
+            addressTypes.put(HOME_FIELD, addressType.getString(ID_FILED));
```

Comment:
We are calling `addressType.getString(ID_FILED)` twice. Can we extract it into a variable?

Reply:
fixed